### ISSUE TITLE
EDSC-825: Temporal filter does not apply to granules outside of temporal range search

### ISSUE THREAD

**Issue by earthdata-tasktop** [nasa/edsc-search-test-config-3#251]
[https://search.earthdata.nasa.gov/search/project/granules?p=C107705237-LPDAAC_ECS&amp;pg](https://search.earthdata.nasa.gov/search/project/granules?p=C107705237-LPDAAC_ECS&pg)[0][qt]=1960-07-18T00%3A00%3A00.000Z%2C2015-07-18T23%3A59%3A59.000Z%2C199%2C199&amp;m=73.60315287691199!-72.26947198845808!0!0!0!&amp;tl=1083141501!5!!&amp;qt=2000-07-18T00%3A00%3A00.000Z%2C2015-07-18T16%3A47%3A59.395Z%2C199%2C199&amp;sp=-24.41249656677246%2C73.80000042915344

In this instance, there are granules listed that have temporal end and start date not overlapping with temporal search range

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun Hi Bo:  ]I was not the reporter of this issue - it was Nemo.  

[~ndjuric]Nemo:  Could you please add a comment on how to reproduce the url in the description?

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

Looks like a CMR issue. Will re-test when CMR-2166 is fixed.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen_carr], could you add a comment on how to reproduce the url in the description?

Thanks!

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun Hi Bo:  ]I was not the reporter of this issue - it was Nemo.  

[~ndjuric]Nemo:  Could you please add a comment on how to reproduce the url in the description?

**Comment by earthdata-tasktop** [comment]
(Comment from Nemanja Djuric):

Follow the link. In the granule list there are granules that are out of search temporal range. I.e.  

MOD13Q1.A2015177.h17v01.005.2015194105359.hdf has  

      BeginningDateTime: 2015-06-26T00:00:00.000000Z  

      EndingDateTime: 2015-07-11T23:59:59.000000Z

There should be only 16 granules or so. If you click on timeline, it will give you different results (that seem to be right).

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

Looks like a CMR issue. Will re-test when CMR-2166 is fixed.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Verify

To verify in OPS, use the link in the description.  

To verify in UAT, use this link:  

[https://search.uat.earthdata.nasa.gov/search/granules?p=C1152-LPDAAC_TS1&amp;amp;m=73.58613399211356!-72.38601946460177!0!0!0!&amp;amp;tl=1083141501!5!!&amp;amp;qt=1999-07-28T00%3A00%3A00.000Z%2C2009-08-03T23%3A59%3A59.000Z%2C209%2C215](https://search.uat.earthdata.nasa.gov/search/granules?p=C1152-LPDAAC_TS1&m=73.58613399211356!-72.38601946460177!0!0!0!&tl=1083141501!5!!&qt=1999-07-28T00%3A00%3A00.000Z%2C2009-08-03T23%3A59%3A59.000Z%2C209%2C215) (other parameters like spatial constraints are omitted)

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

I checked both the OPS and the UAT links provided in the Verify instructions.   In both cases, all granules on the granule list were overlapping or within the temporal range specified.

I then tried my own test in UAT, where I selected a MODIS collection, viewed the granule list, then entered a temporal filter using the Recurring option.   All granules on the resulting filtered list conformed to the temporal filter specified.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Both links (OPS and UAT) are for a temporal filter using the Recurring option, where a date range is specified, and then a separate range of years for those dates.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun Hi Bo:  ]I was not the reporter of this issue - it was Nemo.  

[~ndjuric]Nemo:  Could you please add a comment on how to reproduce the url in the description?

**Comment by earthdata-tasktop** [comment]
(Comment from Nemanja Djuric):

Follow the link. In the granule list there are granules that are out of search temporal range. I.e.  

MOD13Q1.A2015177.h17v01.005.2015194105359.hdf has  

      BeginningDateTime: 2015-06-26T00:00:00.000000Z  

      EndingDateTime: 2015-07-11T23:59:59.000000Z

There should be only 16 granules or so. If you click on timeline, it will give you different results (that seem to be right).

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

Looks like a CMR issue. Will re-test when CMR-2166 is fixed.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Verify

To verify in OPS, use the link in the description.  

To verify in UAT, use this link:  

[https://search.uat.earthdata.nasa.gov/search/granules?p=C1152-LPDAAC_TS1&amp;amp;m=73.58613399211356!-72.38601946460177!0!0!0!&amp;amp;tl=1083141501!5!!&amp;amp;qt=1999-07-28T00%3A00%3A00.000Z%2C2009-08-03T23%3A59%3A59.000Z%2C209%2C215](https://search.uat.earthdata.nasa.gov/search/granules?p=C1152-LPDAAC_TS1&m=73.58613399211356!-72.38601946460177!0!0!0!&tl=1083141501!5!!&qt=1999-07-28T00%3A00%3A00.000Z%2C2009-08-03T23%3A59%3A59.000Z%2C209%2C215) (other parameters like spatial constraints are omitted)

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

I checked both the OPS and the UAT links provided in the Verify instructions.   In both cases, all granules on the granule list were overlapping or within the temporal range specified.

I then tried my own test in UAT, where I selected a MODIS collection, viewed the granule list, then entered a temporal filter using the Recurring option.   All granules on the resulting filtered list conformed to the temporal filter specified.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Both links (OPS and UAT) are for a temporal filter using the Recurring option, where a date range is specified, and then a separate range of years for those dates.

**Comment by earthdata-tasktop** [comment]
(Comment from Nemanja Djuric):

Follow the link. In the granule list there are granules that are out of search temporal range. I.e.  

MOD13Q1.A2015177.h17v01.005.2015194105359.hdf has  

      BeginningDateTime: 2015-06-26T00:00:00.000000Z  

      EndingDateTime: 2015-07-11T23:59:59.000000Z

There should be only 16 granules or so. If you click on timeline, it will give you different results (that seem to be right).

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

Looks like a CMR issue. Will re-test when CMR-2166 is fixed.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Verify

To verify in OPS, use the link in the description.  

To verify in UAT, use this link:  

[https://search.uat.earthdata.nasa.gov/search/granules?p=C1152-LPDAAC_TS1&amp;amp;m=73.58613399211356!-72.38601946460177!0!0!0!&amp;amp;tl=1083141501!5!!&amp;amp;qt=1999-07-28T00%3A00%3A00.000Z%2C2009-08-03T23%3A59%3A59.000Z%2C209%2C215](https://search.uat.earthdata.nasa.gov/search/granules?p=C1152-LPDAAC_TS1&m=73.58613399211356!-72.38601946460177!0!0!0!&tl=1083141501!5!!&qt=1999-07-28T00%3A00%3A00.000Z%2C2009-08-03T23%3A59%3A59.000Z%2C209%2C215) (other parameters like spatial constraints are omitted)

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

I checked both the OPS and the UAT links provided in the Verify instructions.   In both cases, all granules on the granule list were overlapping or within the temporal range specified.

I then tried my own test in UAT, where I selected a MODIS collection, viewed the granule list, then entered a temporal filter using the Recurring option.   All granules on the resulting filtered list conformed to the temporal filter specified.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Both links (OPS and UAT) are for a temporal filter using the Recurring option, where a date range is specified, and then a separate range of years for those dates.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

##### Verify

To verify in OPS, use the link in the description.  

To verify in UAT, use this link:  

[https://search.uat.earthdata.nasa.gov/search/granules?p=C1152-LPDAAC_TS1&amp;amp;m=73.58613399211356!-72.38601946460177!0!0!0!&amp;amp;tl=1083141501!5!!&amp;amp;qt=1999-07-28T00%3A00%3A00.000Z%2C2009-08-03T23%3A59%3A59.000Z%2C209%2C215](https://search.uat.earthdata.nasa.gov/search/granules?p=C1152-LPDAAC_TS1&m=73.58613399211356!-72.38601946460177!0!0!0!&tl=1083141501!5!!&qt=1999-07-28T00%3A00%3A00.000Z%2C2009-08-03T23%3A59%3A59.000Z%2C209%2C215) (other parameters like spatial constraints are omitted)

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

I checked both the OPS and the UAT links provided in the Verify instructions.   In both cases, all granules on the granule list were overlapping or within the temporal range specified.

I then tried my own test in UAT, where I selected a MODIS collection, viewed the granule list, then entered a temporal filter using the Recurring option.   All granules on the resulting filtered list conformed to the temporal filter specified.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Both links (OPS and UAT) are for a temporal filter using the Recurring option, where a date range is specified, and then a separate range of years for those dates.

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

I checked both the OPS and the UAT links provided in the Verify instructions.   In both cases, all granules on the granule list were overlapping or within the temporal range specified.

I then tried my own test in UAT, where I selected a MODIS collection, viewed the granule list, then entered a temporal filter using the Recurring option.   All granules on the resulting filtered list conformed to the temporal filter specified.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Both links (OPS and UAT) are for a temporal filter using the Recurring option, where a date range is specified, and then a separate range of years for those dates.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Both links (OPS and UAT) are for a temporal filter using the Recurring option, where a date range is specified, and then a separate range of years for those dates.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

I checked both the OPS and the UAT links provided in the Verify instructions.   In both cases, all granules on the granule list were overlapping or within the temporal range specified.

I then tried my own test in UAT, where I selected a MODIS collection, viewed the granule list, then entered a temporal filter using the Recurring option.   All granules on the resulting filtered list conformed to the temporal filter specified.
